src/lkl/setup.c

```diff
@@ -755,9 +784,15 @@ static void lkl_mount_disk(
 
 static void lkl_mount_root_disk(struct enclave_disk_config* disk)
 {
+    bool use_overlay = false;
     int err = 0;
-    char mnt_point[] = {"/mnt/vda"};
-    char new_dev_str[] = {"/mnt/vda/dev/"};
+    const char mnt_point_original[] = {"/mnt/vda"};
+    const char mnt_point_overlay[] = {"/mnt/vda-overlay"};
+    const char mnt_point_overlay_upper[] = { "/mnt/vda-overlay-upper" };
+    const char overlay_upper_dir[] = { "/mnt/vda-overlay-upper/upper" };
+    const char overlay_work_dir[] = { "/mnt/vda-overlay-upper/work" };
+    const char* mnt_point = { use_overlay ? mnt_point_overlay : mnt_point_original };
+    const char* new_dev_str = { use_overlay ? "/mnt/vda-overlay/dev/" : "/mnt/vda/dev/" };
```

Comment:
Perhaps it would be cleaner to only do some of this if `use_overlay` is set.

Reply:
I refactored the code, it's now all inside the `if`.